org.eclipse.m2e.core/src/org/eclipse/m2e/core/internal/project/registry/MavenProjectFacade.java

```diff
@@ -59,6 +74,12 @@
 
   public static final String PROP_CONFIGURATORS = MavenProjectFacade.class.getName() + "/configurators";
 
+  private static final String COMPONENTS_PATH = "META-INF/plexus/components.xml";
+
+  private static final String ARTIFACT_HANDLER_ROLE = "org.apache.maven.artifact.handler.ArtifactHandler";
+
+  private final String[] PACKAGING_TYPES_IGNORING = new String[] {"jar", "war", "ear", "ejb", "pom", "maven-plugin"};
```

Comment:
Ideally, the solution shouldn't require such exception and would be generic enough to work without them. (It's not a blocker for a merge though)

Reply:
See my suggestions, I think the overview page should simply use a Set<String> and we should not bother here....